Tests/Particles/SOAParticle/main.cpp

```diff
@@ -0,0 +1,172 @@
+    int const NReal = pc.NStructReal;
+    int const NInt = pc.NStructInt;
+    int const NArrayReal = pc.NArrayReal;
+    int const NArrayInt = pc.NArrayInt;
+
+    using ParticleType = typename T_PC::ParticleType;
+    using ParticleTileDataType = typename T_PC::ParticleTileType::ParticleTileDataType;
+    using RealVector = amrex::PODVector<ParticleReal, Allocator<ParticleReal> >;
+    using IntVector = amrex::PODVector<int, Allocator<int> >;
+    using SPType = typename T_PC::SuperParticleType;
+
+    const int add_num_particles = 5;
+
+    auto& ptile1 = pc.DefineAndReturnParticleTile(0, 0, 0);
+
+    for (int i = 0; i < add_num_particles; ++i)
+    {
+        ptile1.pos(i, 0) = 12.0;
+        ptile1.pos(i, 1) = 12.0;
+        ptile1.pos(i, 2) = 12.0;
+
+        // TODO
+        ptile1.id(i) = 1;
+        ptile1.cpu(i) = 1;
+    }
+    
+    int lev=0;
+    // int numparticles=0;
+    using MyParIter = ParIter_impl<ParticleType, NArrayReal, NArrayInt>;
+    for (MyParIter pti(pc, lev); pti.isValid(); ++pti) {
+        const int np = pti.numParticles();
+        //const auto t_lev = pti.GetLevel();
+        //const auto index = pti.GetPairIndex();
+        // ...
+
+        // preparing access to particle data: AoS
+        //using PType = ImpactXParticleContainer::ParticleType;
+        auto& aos = pti.GetArrayOfStructs();
+        ParticleType* AMREX_RESTRICT aos_ptr = aos().dataPtr();
+
+        // preparing access to particle data: SoA of Reals
+        auto& soa_real = pti.GetStructOfArrays().GetRealData();
+        amrex::ParticleReal* const AMREX_RESTRICT part_x = soa_real[0].dataPtr();
+        amrex::ParticleReal* const AMREX_RESTRICT part_y = soa_real[1].dataPtr();
+        amrex::ParticleReal* const AMREX_RESTRICT part_z = soa_real[2].dataPtr();
+        amrex::ParticleReal* const AMREX_RESTRICT part_aaa = soa_real[3].dataPtr();
+        auto& soa_int = pti.GetStructOfArrays().GetIntData();
+
+        // Iterating over old Particles
+        ParallelFor( np, [=] AMREX_GPU_DEVICE (long ip)
+        {
+            ParticleType& AMREX_RESTRICT p = aos_ptr[ip];
+            p.pos(0) += 1;
+            p.pos(1) += 1;
+            p.pos(2) += 1;
+            
+            amrex::ParticleReal & AMREX_RESTRICT x = part_x[ip];
+            amrex::ParticleReal & AMREX_RESTRICT y = part_y[ip];
+            amrex::ParticleReal & AMREX_RESTRICT z = part_z[ip];
+            amrex::ParticleReal & AMREX_RESTRICT a = part_aaa[ip];
+
+            x += 1.0;
+            y += 1.0;
+            z += 1.0;
+            a += 1.0;
+        });
+
+        // Iterating over SoA Particles
+        ParticleTileDataType ptd = pti.GetParticleTile().getParticleTileData(); 
+
+        ParallelFor( np, [=] AMREX_GPU_DEVICE (long ip)
+        {
+            ParticleType p(ptd, ip);
+            p.pos(0) += 1;
+            p.pos(1) += 1;
+            p.pos(2) += 1;
+        });
+
+
+    }
+
+    auto tmp = pc.template make_alike<amrex::PinnedArenaAllocator>();
```

Comment:
This is still missing to fill the `tmp` particle container with data:
```suggestion
// create a host-side particle buffer

    //  copy device-to-host
    bool const local = true;
    tmp.copyParticles(pc, local);

```

We need to add this here and fix the compile for it.